src/components/ui/sidebar.tsx

```diff
@@ -112,6 +112,10 @@ const SidebarProvider = React.forwardRef<
       toggleSidebar,
     );
 
+    React.useEffect(() => {
+      _setOpen(defaultOpen);
+    }, [defaultOpen]);
```

Comment:
the whole meaning of "default" is lost :)

Reply:
@rithviknishad I know, but as we are overriding the opening behaviour, and we want to stick to using the defaultOpen prop, this is the only way I could see 🫠

Reply:
1. why do we even need a useEffect to read the cookie? 🤔 
2. why not just cleanly use localStorage? 🤔 we have 0 cookie usages as of now in care;

Reply:
i don't see any reason for having a useEffect and a useState to read the cookie in the first place, which is why you end up needing to hack the sidebar to get it working

Reply:
i wonder why shadcn is even writing the sidebar's state to cookie but not even utilising it for persistence in the first place;